instrumentation/httpurlconnection/testing/src/androidTest/kotlin/io/opentelemetry/instrumentation/library/httpurlconnection/InstrumentationTest.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.library.httpurlconnection
+
+import io.opentelemetry.android.test.common.OpenTelemetryTestUtils
+import io.opentelemetry.instrumentation.library.httpurlconnection.HttpUrlConnectionTestUtil.executeGet
+import io.opentelemetry.instrumentation.library.httpurlconnection.HttpUrlConnectionTestUtil.post
+import io.opentelemetry.instrumentation.library.httpurlconnection.internal.HttpUrlConnectionSingletons
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter
+import org.junit.Assert
+import org.junit.Test
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+class InstrumentationTest {
+    @Test
+    fun testHttpUrlConnectionGetRequest_ShouldBeTraced() {
+        val inMemorySpanExporter = InMemorySpanExporter.create()
```

Comment:
This `inMemorySpanExporter` is created at the beginning of every test method and shutdown at the bottom of each. I think this can be better handled by making the exporter a class field and creating it in a `@BeforeEach` method and shutting it down in the `@AfterEach` method.